proxy/hdrs/VersionConverter.cc

```diff
@@ -0,0 +1,265 @@
+/** @file
+
+  HTTP header version converter
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include "VersionConverter.h"
+#include "tscpp/util/LocalBuffer.h"
+
+int
+VersionConverter::convert(HTTPHdr &header, int from, int to) const
+{
+  int type;
+
+  switch (http_hdr_type_get(header.m_http)) {
+  case HTTP_TYPE_REQUEST:
+    type = 0;
+    break;
+  case HTTP_TYPE_RESPONSE:
+    type = 1;
+    break;
+  default:
```

Comment:
I suggest changing from `default:` to `case HTTP_TYPE_UNKNOWN:`. That way if, for some reason, another enumeration is added, the compiler will warn about an unhandled case here which will remind the person changing the enum to check the logic in this switch.